presto-tpch/src/main/java/com/facebook/presto/tpch/statistics/StatisticsEstimator.java

```diff
@@ -0,0 +1,250 @@
+        ));
+    }
+
+    private ColumnStatisticsData rescale(TpchColumn<?> column, ColumnStatisticsData big, ColumnStatisticsData small, double scaleFactor)
+    {
+        if (UNSUPPORTED_COLUMNS.contains(column)) {
+            return ColumnStatisticsData.empty();
+        }
+        else {
+            return rescale(big, small, scaleFactor);
+        }
+    }
+
+    private ColumnStatisticsData rescale(ColumnStatisticsData big, ColumnStatisticsData small, double scaleFactor)
+    {
+        if (columnDoesNotScale(big, small)) {
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount(),
+                    big.getNullsCount(),
+                    checkPresent(withBoth(big.getLowestValue(), small.getLowestValue(), this::checkClose)),
+                    checkPresent(withBoth(big.getHighestValue(), small.getHighestValue(), this::checkClose))
+            );
+        }
+        else {
+            Function<Number, Double> rescale = value -> value.doubleValue() * scaleFactor;
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount().map(rescale).map(Number::longValue),
+                    big.getNullsCount().map(rescale).map(Number::longValue),
+                    Types.tryCast(big.getLowestValue(), Number.class),
+                    Types.tryCast(big.getHighestValue(), Number.class).map(rescale));
+        }
+    }
+
+    private boolean columnDoesNotScale(ColumnStatisticsData big, ColumnStatisticsData small)
+    {
+        return withBoth(small.getLowestValue(), big.getLowestValue(), this::areClose)
+                .flatMap(lowerBoundsClose ->
+                        withBoth(small.getHighestValue(), big.getHighestValue(), this::areClose)
+                                .map(upperBoundsClose -> lowerBoundsClose && upperBoundsClose))
+                .orElse(false);
+    }
+
+    private Object checkClose(Object leftValue, Object rightValue)
+    {
+        checkArgument(
+                areClose(leftValue, rightValue),
+                format("Values must be close to each other, got [%s] and [%s]", leftValue, rightValue));
+        return leftValue;
+    }
+
+    private boolean areClose(Object leftValue, Object rightValue)
+    {
+        checkSameType(leftValue, rightValue);
+        if (leftValue instanceof String) {
+            return leftValue.equals(rightValue);
+        }
+        else {
+            Number left = checkType(leftValue, Number.class);
+            Number right = checkType(rightValue, Number.class);
+            return areClose(left.doubleValue(), right.doubleValue());
+        }
+    }
+
+    private boolean areClose(double left, double right)
+    {
+        return abs(right - left) <= abs(left) * 0.01;
+    }
+
+    private TableStatisticsData readStatistics(TpchTable<?> table, Predicate<Map<TpchColumn<?>, Object>> constraint, String schemaName)
+    {
+        if (table.equals(ORDERS) && !requiresAllOrderPartitions(constraint)) {
```

Comment:
you could pass `TupleDomain` to `StatisticsEstimator` with `ValuesSet` (discreet partitioning values) for columns that we support. This way you won't have to have ORDER_STATUS specific code here (`StatisticsEstimator` would be independent of what columns we do support).